packages/ui/src/app/pages/Forum/ForumCategory.tsx

```diff
@@ -48,17 +48,13 @@ export const ForumCategory = () => {
     },
     { perPage: THREADS_PER_PAGE, page }
   )
-  const isRefetched = useRefetchQueries({
+  useRefetchQueries({
     interval: MILLISECONDS_PER_BLOCK,
     include: ['GetForumThreads', 'GetForumThreadsCount'],
   })
 
   const { showModal } = useModal()
 
-  if (isLoadingThreads && !isRefetched) {
-    return <Loading />
-  }
-
```

Comment:
Great work on catching this ! It wasn't an easy one to debug :exploding_head: 

But I think the fix could be improved because first it now makes empty categories show a loader every 6 seconds e.g: https://dao-git-fork-mkbeefcake-fix-4485-4484-joystream.vercel.app/#/forum/category/17?network-config=https://34.230.5.182.nip.io/network/config.json. But more importantly it still refetch way to much data (as you pointed out on #4484). Also I realized we've struggled with this particular `useRefetch` for a long time (see #4318) so I really think the best is to remove this one.

Instead we could replace this logic by a `pollInterval` of 6 second in `packages/ui/src/forum/hooks/useForumCategoryThreads.ts` but only when `isArchive` is `false`. Also `isLoading` should become `!threadsData || !countData` this way `isLoading` is only `true` on the initial load.

Reply:
![image](https://github.com/Joystream/pioneer/assets/69617937/391695d4-bda2-4a34-8cdc-4818da90f83d)
Thanks for your guide. also I figured out "posts" field in forumThreads query, If i choose this mechanism, i can reduce the query count to 1, but it could take all posts info per each thread, so could take much data and a bit more loading time.
I hope to get your opinion about this.

Reply:
Yes `forumThreads { posts { ... } }` will return all the posts for all the threads shown one the category page so I think 1 request per thread is a better trade off.

Reply:
@thesan I have added posts field for ForumThread interface and fetched it in GetForumThreads Query, so reduce the queries in Forum Category page, please check the updated code.